updated aws4 version and fixed the  Authentication issue

Jay07GIT opened the pull request:
Issue: #147
Changes in this PR.

1. Updated `aws4` version
2. Fixed the Authentication issue caused by `axios` dependency.

To test.
Create a test.py file in the local with the below code and run `node test.py`

```
const VinylDNS = require('./');
const vinylClient = new VinylDNS({
  apiUrl: 'https://vinyldns.example.com:9443',
  accessKeyId: 'testAccesskey',
  secretAccessKey: 'testSecretKey'
});

vinylClient.getGroup('global-acl-group-id')
.then(result => {
  console.log(result);
}).catch(err => {
  console.log(err);
});
```

Jay07GIT committed:
updated aws4 version and fixed the  Authentication issue

Jay07GIT committed:
update package-json

Jay07GIT committed:
update

Jay07GIT committed:
update

Jay07GIT committed:
update

Jay07GIT committed:
resolved failed tests

Jay07GIT committed:
update

Aravindh-Raju commented on src/request.js:
```diff
@@ -56,7 +56,7 @@ class Request {
         'Content-Type': 'application/json'
       },
-      body: opts.body ? JSON.stringify(opts.body) : ''
+      body: JSON.stringify(opts.body)
```
Will we get an error when opts.body is empty as the condition is removed?

Jay07GIT replied:
Yeh . If the body is null or undefined, we will get an error.

nspadaccino replied:
I'm not clear on why this change is required, Jay could you explain?

Aravindh-Raju commented on bin/run-integration-tests.sh:
```diff
@@ -5,7 +5,7 @@ set -e
 
 echo "Removing VinylDNS API if exists.."
-if docker ps | grep -q "vinyldns-js-api"; then
+if docker ps -a| grep -q "vinyldns-js-api"; then
```
I think the `-a` is added for testing? If so, we can remove them.

Jay07GIT replied:
this done purposely to pull all running docker container and stop which has the name `vinyldns-js-api`

Aravindh-Raju reviewed (approved):
Tested it out locally. Able to perform all the actions now. LGTM!

Jay07GIT committed:
Merge branch 'master' into axios_issue_fix

Jay07GIT committed:
updated axios version to 1.13.6

Signed-off-by: Jay07GIT <jeyraj931@gmail.com>

Jay07GIT committed:
updated sync tests

Signed-off-by: Jay07GIT <jeyraj931@gmail.com>

JoshSEdwards reviewed (dismissed):
Changes look good, approved.

nspadaccino reviewed (commented):
Is aws4 actually being updated here? I don't see that in package.json. If not, we should update the PR title and description.

Jay07GIT committed:
updated aws4 version in package.json

Signed-off-by: Jay07GIT <jeyraj931@gmail.com>

Jay07GIT commented:
> Is aws4 actually being updated here? I don't see that in package.json. If not, we should update the PR title and description.

@nspadaccino Initially i have updated the aws4 version, but in later [merge commit](https://github.com/vinyldns/vinyldns-js/pull/142/changes/d22b62e421e280f0f5d950f28cdf5cf5ba2b7ec1) from the main, version has been revoked which i have not noticed. Thanks for finding. Updated the version in the latest commit.

JoshSEdwards reviewed (dismissed):
LGTM

nspadaccino commented on test/vinyldns_test.js:
```diff
@@ -167,8 +167,6 @@ describe('VinylDNS', () => {
           .then(result => {
             assert.equal(result.zone.name, 'sync-test.');
-
-            done();
```
When a test:                                                                                                     
  1. Has no `done` callback parameter                                                                                                                             
  2. Does not return the promise                                                                                                                                
                                                                                                                                                                
Mocha considers the test synchronous and passes it immediately, before the promise resolves. 
So these tests will always pass regardless of what the API returns.    

This can be fixed by changing `vinyl.syncZone('123') ` to ` return vinyl.syncZone('123') `

Jay07GIT replied:
Updated the code and description

JoshSEdwards replied:
I updated the description to remove the real API endpoint

nspadaccino reviewed (changes requested):
Couple comments. Also, the description for this PR should be edited to remove the real endpoint and credentials, as this is an open source repo.

Jay07GIT committed:
update

Signed-off-by: Jay07GIT <jeyraj931@gmail.com>

nspadaccino reviewed (approved):
LGTM
